python_modules/libraries/dagster-sigma/dagster_sigma/resource.py

```diff
@@ -426,6 +426,18 @@ def load_sigma_asset_specs(organization: SigmaOrganization) -> Sequence[AssetSpe
         )
 
 
+def _get_translator_spec_assert_keys_match(
+    translator: DagsterSigmaTranslator, data: Union[SigmaDataset, SigmaWorkbook]
+) -> AssetSpec:
+    key = translator.get_asset_key(data)
+    spec = translator.get_asset_spec(key, data)
+    check.invariant(
+        spec.key == key,
+        f"Key on AssetSpec returned by {translator.__class__.__name__}.get_asset_spec {spec.key} does not match input key {key}",
+    )
```

Comment:
Might be called a lot so recommend putting this in an `if` so that the string is only constructed on failure.